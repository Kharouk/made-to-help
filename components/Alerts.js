import * as firebase from 'firebase';
import React from 'react';
import {
  View, TextInput, Text, Button, Alert,
} from 'react-native';
import { Location } from 'expo';

export default class Alerts extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      longitude: null,
      latitude: null,
      text: '',
      typeOfReport: '',
    };
  }

  componentWillMount = async () => {
    const location = await Location.getCurrentPositionAsync({});
    this.setState({ longitude: location.coords.longitude, latitude: location.coords.latitude });
  };

  writeAlertData = (body, location, typeOfReport) => {
    firebase
      .database()
      .ref('alerts/')
      .push({
        body,
        location,
        typeOfReport,
      })
      .then((data) => {
        console.log('data ', data);
      })
      .catch((err) => {
        console.log('Error: ', err);
      });
    Alert.alert('Thanks, your report has been submitted. ðŸ™ƒ');
  };

  sendData = () => {
    this.writeAlertData(this.state.text, {
      latitude: this.state.latitude,
      longitude: this.state.longitude,
    },
    this.state.typeOfReport
  );
    this.setState({ text: '', typeOfReport: '' });
    this.navigateHome();
  };

  navigateHome = () => {
    this.props.navigation.navigate('Home');
  }

  onHandleChange = (event) => {
    this.setState({ text: event });
  }

  render() {
    return (
      <View>
        <Text>Inaccessibility Report </Text>
        <Button title="Broken Lift" onPress={ () => this.setState({typeOfReport: 'Broken Lift'})}
          />
        <TextInput placeholder="Extra Info" onChangeText={this.onHandleChange}>
          {this.state.text}
        </TextInput>
        {this.state.text !== '' || this.state.typeOfReport !== '' && <Button title="Submit" onPress={this.sendData} />}
      </View>
    );
  }
}
