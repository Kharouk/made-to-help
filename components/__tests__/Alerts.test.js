import React from 'react';
import renderer from 'react-test-renderer';
import { TextInput, Button } from "react-native";
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Alerts from '../Alerts';

Enzyme.configure({ adapter: new Adapter() });

test('Renders CurrentLocation component', () => {
  const snapshot = shallow(<Alerts />);
  expect(snapshot).toMatchSnapshot();
});

test('Input changes', () => {
  const snapshot = shallow(<Alerts />);
  snapshot.find('#show-form').simulate('press')
  userInput = snapshot.find(TextInput)
  userInput.simulate('changeText', 'Lift broken')
  expect(snapshot.state().text).toEqual('Lift broken')
});

test('Sending report data for broken lift', () => {
  const snapshot = shallow(<Alerts />);
  snapshot.find('#brokenLift').simulate('press')
  expect(snapshot.state().typeOfReport).toEqual('Broken Lift')
});

test('Sending report data for blocked path', () => {
  const snapshot = shallow(<Alerts />);
  snapshot.find('#blockedPath').simulate('press')
  expect(snapshot.state().typeOfReport).toEqual('Blocked Path')
});

test('Sending report data for stairs', () => {
  const snapshot = shallow(<Alerts />);
  snapshot.find('#stairs').simulate('press')
  expect(snapshot.state().typeOfReport).toEqual('Stairs')
});

test('Sending report data for narrow path', () => {
  const snapshot = shallow(<Alerts />);
  snapshot.find('#narrowPath').simulate('press')
  expect(snapshot.state().typeOfReport).toEqual('Narrow Path')
});

test('Sending report data for No Ramp', () => {
  const snapshot = shallow(<Alerts />);
  snapshot.find('#noRamp').simulate('press')
  expect(snapshot.state().typeOfReport).toEqual('No Ramp')
});
