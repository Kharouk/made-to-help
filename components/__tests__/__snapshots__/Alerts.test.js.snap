// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CurrentLocation component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Alerts />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component>
          Inaccessibility Report
        </Component>,
        <TouchableOpacity
          activeOpacity={0.2}
          id="brokenLift"
          onPress={[Function]}
        >
          <Component
            source={1}
          />
          <Component>
            Broken Lift
          </Component>
        </TouchableOpacity>,
        <TouchableOpacity
          activeOpacity={0.2}
          id="blockedPath"
          onPress={[Function]}
        >
          <Component
            source={1}
          />
          <Component>
            Blocked Path
          </Component>
        </TouchableOpacity>,
        <TouchableOpacity
          activeOpacity={0.2}
          id="stairs"
          onPress={[Function]}
        >
          <Component
            source={1}
          />
          <Component>
            Stairs
          </Component>
        </TouchableOpacity>,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Component
            source={1}
          />
          <Component>
            Narrow Path
          </Component>
        </TouchableOpacity>,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Component
            source={1}
          />
          <Component>
            No Ramp
          </Component>
        </TouchableOpacity>,
        <TouchableOpacity
          activeOpacity={0.2}
          id="show-form"
          onPress={[Function]}
        >
          <Component
            source={1}
          />
        </TouchableOpacity>,
        <Component>
          Other
        </Component>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Inaccessibility Report",
        },
        "ref": null,
        "rendered": "Inaccessibility Report",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": Array [
            <Component
              source={1}
            />,
            <Component>
              Broken Lift
            </Component>,
          ],
          "id": "brokenLift",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Broken Lift",
            },
            "ref": null,
            "rendered": "Broken Lift",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": Array [
            <Component
              source={1}
            />,
            <Component>
              Blocked Path
            </Component>,
          ],
          "id": "blockedPath",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Blocked Path",
            },
            "ref": null,
            "rendered": "Blocked Path",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": Array [
            <Component
              source={1}
            />,
            <Component>
              Stairs
            </Component>,
          ],
          "id": "stairs",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Stairs",
            },
            "ref": null,
            "rendered": "Stairs",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": Array [
            <Component
              source={1}
            />,
            <Component>
              Narrow Path
            </Component>,
          ],
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Narrow Path",
            },
            "ref": null,
            "rendered": "Narrow Path",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": Array [
            <Component
              source={1}
            />,
            <Component>
              No Ramp
            </Component>,
          ],
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "No Ramp",
            },
            "ref": null,
            "rendered": "No Ramp",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Component
            source={1}
          />,
          "id": "show-form",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Other",
        },
        "ref": null,
        "rendered": "Other",
        "type": [Function],
      },
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component>
            Inaccessibility Report
          </Component>,
          <TouchableOpacity
            activeOpacity={0.2}
            id="brokenLift"
            onPress={[Function]}
          >
            <Component
              source={1}
            />
            <Component>
              Broken Lift
            </Component>
          </TouchableOpacity>,
          <TouchableOpacity
            activeOpacity={0.2}
            id="blockedPath"
            onPress={[Function]}
          >
            <Component
              source={1}
            />
            <Component>
              Blocked Path
            </Component>
          </TouchableOpacity>,
          <TouchableOpacity
            activeOpacity={0.2}
            id="stairs"
            onPress={[Function]}
          >
            <Component
              source={1}
            />
            <Component>
              Stairs
            </Component>
          </TouchableOpacity>,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <Component
              source={1}
            />
            <Component>
              Narrow Path
            </Component>
          </TouchableOpacity>,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <Component
              source={1}
            />
            <Component>
              No Ramp
            </Component>
          </TouchableOpacity>,
          <TouchableOpacity
            activeOpacity={0.2}
            id="show-form"
            onPress={[Function]}
          >
            <Component
              source={1}
            />
          </TouchableOpacity>,
          <Component>
            Other
          </Component>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Inaccessibility Report",
          },
          "ref": null,
          "rendered": "Inaccessibility Report",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": Array [
              <Component
                source={1}
              />,
              <Component>
                Broken Lift
              </Component>,
            ],
            "id": "brokenLift",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Broken Lift",
              },
              "ref": null,
              "rendered": "Broken Lift",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": Array [
              <Component
                source={1}
              />,
              <Component>
                Blocked Path
              </Component>,
            ],
            "id": "blockedPath",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Blocked Path",
              },
              "ref": null,
              "rendered": "Blocked Path",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": Array [
              <Component
                source={1}
              />,
              <Component>
                Stairs
              </Component>,
            ],
            "id": "stairs",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Stairs",
              },
              "ref": null,
              "rendered": "Stairs",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": Array [
              <Component
                source={1}
              />,
              <Component>
                Narrow Path
              </Component>,
            ],
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Narrow Path",
              },
              "ref": null,
              "rendered": "Narrow Path",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": Array [
              <Component
                source={1}
              />,
              <Component>
                No Ramp
              </Component>,
            ],
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "No Ramp",
              },
              "ref": null,
              "rendered": "No Ramp",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Component
              source={1}
            />,
            "id": "show-form",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Other",
          },
          "ref": null,
          "rendered": "Other",
          "type": [Function],
        },
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
